aria_plugin/environment.py

```diff
@@ -0,0 +1,112 @@
+import errno
+import os
+import shutil
+
+import aria
+from aria.core import Core
+from aria.orchestrator.plugin import PluginManager
+from aria.storage.sql_mapi import SQLAlchemyModelAPI
+from aria.storage.filesystem_rapi import FileSystemResourceAPI
+from cloudify import ctx
+
+
+class Environment(object):
+
+    MANAGER_RESOURCES_DIR = '/opt/manager/resources'
+    BLUEPRINTS_DIR = os.path.join(MANAGER_RESOURCES_DIR, 'blueprints')
+    CLOUDIFY_PLUGINS_DIR = os.path.join(MANAGER_RESOURCES_DIR, 'plugins')
+
+    def __init__(self):
+
+        self._mk_working_dir()
+
+        self._to_clean = []
+
+        # Initialized lazily:
+        self._model_storage = None
+        self._resource_storage = None
+        self._plugin_manager = None
+        self._core = None
+
+    @property
+    def workdir(self):
+        return self._workdir
+
+    @property
+    def blueprint_dir(self):
+        return os.path.join(self.BLUEPRINTS_DIR,
```

Comment:
consider removing and using the `ctx.get_resource` (or something similar)

Reply:
Since the csar_path property can be either url or a path, and since the csar module already handles csar urls, using ctx.get_resource would be cumbersome here.